solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -745,6 +754,47 @@ contract InterchainAccountRouter is Router {
             );
     }
 
+    function sendCommitment(
+        uint32 _destination,
+        bytes32 _router,
+        bytes32 _ism,
+        bytes memory _hookMetadata,
+        IPostDispatchHook _hook,
+        bytes32 _salt,
+        bytes32 _commitment
+    ) public payable returns (bytes32) {
+        bytes memory _body = InterchainAccountMessage.encodeCommitment({
+            _owner: msg.sender.addressToBytes32(),
+            _ism: _ism,
+            _commitment: _commitment,
+            _userSalt: _salt
+        });
+        return
+            _dispatchMessageWithHook(
+                _destination,
+                _router,
+                _ism,
+                _body,
+                _hookMetadata,
+                _hook
+            );
+    }
+
+    error InvalidCommitment(OwnableMulticall ICA, bytes32 badCommitment);
```

Comment:
prefer revert strings to custom errors
we can consider doing this migration wholesale in future but have found that custom error decoding is not universally supported across client side tooling